msal/application.py

```diff
@@ -551,6 +551,13 @@ def acquire_token_silent_with_error(
             return result
         final_result = result
         for alias in self._get_authority_aliases(self.authority.instance):
+            if not self.token_cache.find(
+                    self.token_cache.CredentialType.REFRESH_TOKEN,
+                    target=scopes,
+                    query={"environment": alias}):
+                # Skip heavy weigh logic when RT for this alias doesn't exist.
```

Comment:
@abhidnya13 
I went ahead and made some changes, but github does not allow me to set you as reviewer (presumably because you were the PR creator). Please leave your review as comments here.

Reply:
This would also skip finding an AT if there exists one though which is done in line 594

Reply:
True. The reasoning here is to use a simplest-possible, good-enough logic as a heuristic, rather than duplicates most of the real `acquire_token_silent(...)` logic here. Given that RT basically always exists in an active alias, this is probably a good-enough approach to decide whether an alias is worth further searching.

Reply:
That is true that this is a good enough approach given the reason you mentioned. I think we can merge this in now to provide a quick fix but may be this is worth exploring and putting some more time to refactor the code sometime later to implement a foolproof approach instead which won't skip finding an AT